draco/analysis/powerspec.py

```diff
@@ -0,0 +1,1504 @@
+
+    For each delay channel, it will keep the non-zero visibilities between min and max
+    baseline length and flatten the array, i.e, the output is of shape (nvis),
+    where nvis is number of vis between two limits.
+
+    Parameters
+    ----------
+    data_cube : np.ndarray[kx,ky]
+     The data cube to reshape and flatten.
+    u : np.ndarray[kx]
+     The u-coordinates in wavelength unit.
+    v : np.ndarray[ky]
+     The v-coordinates in wavelength unit.
+    bl_min : float
+     The min baseline length in wavelength unit.
+    bl_max : float
+     The max baseline length in wavelength unit.
+
+    Returns
+    -------
+    ft_cube : np.ndarray[nvis]
+     The flatten data cube.
+    uu : np.ndarray[nvis]
+     The flatten u-coordinates
+    vv : np.ndarray[nvis]
+     The flatten v-coordinates
+    """
+    g_uu, g_vv = np.meshgrid(v, u)
+    g_ru = np.sqrt(g_uu**2 + g_vv**2)
+    bl_idx = (g_ru >= bl_min) & (g_ru <= bl_max)
+    uu = g_uu[bl_idx]
+    vv = g_vv[bl_idx]
+    ft_cube = data_cube[bl_idx].flatten()
+
+    return ft_cube, uu.flatten(), vv.flatten()
+
+
+def get_2d_ps(ps_cube, weight, kperp_bins, uu, vv, redshift, cosmo=None):
+    """Estimate the cylindrically averaged 2D power spectrum.
+
+    Parameters
+    ----------
+    ps_cube : np.ndarray[nbl]
+      The power spectrum array to average in cylindrical bins.
+    weight : np.ndarray[nbl]
+      The weight to be used in averaging.
+       If None, then use unit unifrom weight.
+    kperp_bins : float
+      The kperp values of each bin, at which the power spectrum will be
+       calculated. Unit: Mpc^-1
+    uu : np.ndarray[u]
+      The flatten u-coordinate in wavelength.
+    vv : np.ndarray[v]
+      The flatten v-coordinate in wavelength.
+    redshift : float
+      The redshift corresponding to the band center.
+    cosmo: Cosmology object
+        Default is cora.util.Cosmology() default.
+
+    Returns
+    -------
+    ps_2D : np.ndarray[kperp]
+      The  binned  power along k_perp (cylindrical binning).
+
+    ps_2D_w : np.ndarray[kperp]
+      The binned  weight along k_perp (cylindrical binning).
+
+    """
+    if cosmo is None:
+        cosmo = get_cosmo()
+
+    # Weight for inverse variance weighted avg
+    w = weight
+
+    # Find the bin indices and determine which radial bin each cell lies in.
+    ku = u_to_kperp(uu, redshift, cosmo=cosmo)  # convert the u-array to fourier modes
+    kv = u_to_kperp(vv, redshift, cosmo=cosmo)  # convert the v-array to fourier modes
+    ru = np.sqrt(ku**2 + kv**2)
+
+    # Digitize the bins
+    bin_indx = np.digitize(ru, bins=kperp_bins)
+
+    # Define empty list to store the binned 2D PS
+    ps_2D = []
+    ps_2D_w = []
+
+    # Now bin in 2D ##
+    with np.errstate(divide="ignore", invalid="ignore"):
+        for i in np.arange(len(kperp_bins) - 1) + 1:
+            p = np.nansum(w[bin_indx == i] * ps_cube[bin_indx == i]) / np.sum(
```

Comment:
```suggestion
            p = np.sum(w[bin_indx == i] * ps_cube[bin_indx == i]) / np.sum(
```
Same reason as above.

Reply:
Done.